### PR TITLE
Remove legacy nodeagent/filemount config from agent

### PR DESCRIPTION
**Pull request by howardjohn**
For https://github.com/istio/istio/issues/21834
For https://github.com/istio/istio/issues/21222

This cleans up a bunch of old code around handling file mounted certs
and node agent SDS. File mounts are still supported, but are served over
SDS.

This will have merge conflicts with
https://github.com/istio/istio/pull/22105 but otherwise is compatible
with that change.

### CODE REVIEW

File: tools/istio-docker.mk [hunk 1]
```diff
@@ -79,9 +79,7 @@ docker.proxyv2: install/gcp/bootstrap/gcp_envoy_bootstrap.json
 docker.proxyv2: $(ISTIO_ENVOY_LINUX_RELEASE_DIR)/envoy
 docker.proxyv2: $(ISTIO_OUT_LINUX)/pilot-agent
 docker.proxyv2: pilot/docker/Dockerfile.proxyv2
-docker.proxyv2: pilot/docker/envoy_pilot.yaml.tmpl
```
Comment (howardjohn) [inline tools/istio-docker.mk]: these are not strictly related to this change, but these are neither used nor do they work anymore. I cleaned them up since they reference the SDS token field I removed

File: pilot/docker/envoy_telemetry.yaml.tmpl [hunk 2]
```diff
@@ -1,262 +0,0 @@
-admin:
```
Comment (kyessenov) [inline pilot/docker/envoy_telemetry.yaml.tmpl]: Where is this file now?

Comment (howardjohn) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: https://github.com/istio/istio/blob/6fab76ef7ba34a9e551e0313b67371064cc3a522/manifests/istio-telemetry/mixer-telemetry/templates/configmap-envoy.yaml#L5

Comment (kyessenov) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: I'm not sure how you can get internal Mixer telemetry without using a custom roller bootstrap.

Comment (howardjohn) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: We still have a custom bootstrap

Comment (kyessenov) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: I see. Although I'm not sure I agree to place it in the installation. I think mixer codebase would be the appropriate place for its bootstrap.

File: tools/istio-docker.mk [hunk 3]
```diff
@@ -79,9 +79,7 @@ docker.proxyv2: install/gcp/bootstrap/gcp_envoy_bootstrap.json
...
 docker.proxyv2: pilot/docker/envoy_policy.yaml.tmpl
```
Comment (kyessenov) [inline tools/istio-docker.mk]: Why not clean up policy one, too? Telemetry and policy bootstraps should be next to each other.

Comment (howardjohn) [reply tools/istio-docker.mk]: Policy is still using this for whatever reason. I don't know why, they clearly should be consistent. I don't want to do that in this PR though. I can keep the pilot/telemetry file around until then, but they are completely broken and not used which is why I removed them

File: pilot/docker/envoy_telemetry.yaml.tmpl (hunk 2 above)
Comment (kyessenov) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: In general, installation files have extremely poor coverage since each test ends up being e2e. If we can separate resources away from k8s/install, then we can test them locally in a unit test without running istio or even k8s.

Comment (howardjohn) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: I don't necessarily disagree but this is completely unrelated to the PR, can you open an issue if you want to change this?

File: tools/istio-docker.mk (hunk 3 above)
Comment (kyessenov) [reply tools/istio-docker.mk]: We had a test for internal telemetry, so it should be still working, right? We have to keep mixer features until 1.7 since it's still supported.

Comment (howardjohn) [reply tools/istio-docker.mk]: Yes, but pilot/docker/envoy_telemetry.yaml.tmpl is never used, because we are using the file in manifests/ I linked about. This is dead code I am removing, nothing more

File: pilot/docker/envoy_telemetry.yaml.tmpl (hunk 2 above)
Comment (kyessenov) [reply pilot/docker/envoy_telemetry.yaml.tmpl]: Well, I'm not the one moving them :) I'd rather leave things the way they were even if it completely makes no sense (which I agree with). Otherwise, we have to make them better, and it's not worth it.

**Comment by howardjohn** [comment]
I think this is blocked by https://github.com/istio/istio/pull/21658

File: tools/istio-docker.mk (hunk 3 above)
Comment (kyessenov) [reply tools/istio-docker.mk]: Ok. In any case, you don't need my approval for this PR.

File: pilot/cmd/pilot-agent/config.go [hunk 4]
```diff
@@ -30,6 +32,29 @@ import (
 	"istio.io/pkg/log"
 )
 
+// getTLSCerts returns all file based certificates from mesh config
+// TODO(https://github.com/istio/istio/issues/21834) serve over SDS instead of files
+func getTLSCerts(pc meshconfig.ProxyConfig) []string {
+	certs := []string{}
+	appendTLSCerts := func(rs *meshconfig.RemoteService) {
```
Comment (costinm) [inline pilot/cmd/pilot-agent/config.go]: More comments please. 

What are they used for ? How are they distributed to sidecars ? Do we still need them - i.e. stable/beta features we support ? Can we use a generic API ( like Sidecar with Shriram's new TLS support ) to generalize the load of certs we need for various purpose ? 

At least as TODO - I understand you are just cleaning up.

Comment (howardjohn) [reply pilot/cmd/pilot-agent/config.go]: added some more context. they are file mounted though

File: tests/testdata/certs/default/cert-chain.pem [hunk 5]
```diff
@@ -1,19 +1,20 @@
...
-Vd2whWSI8iQpPlEpz3soYliptO2Y/W3TssMKIExLAgMBAAGjcjBwMAkGA1UdEwQC
-MAAwCwYDVR0PBAQDAgXgMB0GA1UdJQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATA3
-BgNVHREEMDAuhixzcGlmZmU6Ly9jbHVzdGVyLmxvY2FsL25zL2RlZmF1bHQvc2Ev
-ZGVmYXVsdDANBgkqhkiG9w0BAQsFAAOCAQEA1Glwoy03FBk36RmadFj6REJ6CR+O
-tJqqUPu6JLeT2ohKUZdC7Ns/CHm8c74rI0HuzDOIgmCjqQ3XY3MeGibp6zERzKFA
-2t7LPRzzRO2TFOkPC8pxUCfIqQ/WYjfL2HRsN//YoJa2BzrMHAah3HimQFTS+Kj0
-6iuKHnmBktrVgl/mdcsAU9cCPIfEg3aJBfQCbuAFVxIf5IXD7z65xiR/fpSh1EQz
-dZbKy8+GLzjUqi3uxFkxNqSPhLuA9K2QzZCp1C206TGQ7GR8CFcMLPYTAtV/TgPB
-9RSmjRU7ihdBinKfF7BXW3XkQ2mICngQFW51MKm4Pys3r7VDRy+dRl8/dg==
+MIIDMjCCAhqgAwIBAgIUaAVaH1ssljSp51Hm0bAEJUVf4dMwDQYJKoZIhvcNAQEL
```
Comment (howardjohn) [inline tests/testdata/certs/default/cert-chain.pem]: Needed to recreate the certs to add `istiod` as part of the SAN, before it was only pilot. I also included the generate.sh script that was used to create that, before it was ad hoc

**Comment by howardjohn** [comment]
/retest

don't understand why it fails.. it says all tests passed

**Comment by howardjohn** [comment]
@nmittler you are probably the most familiar in T&R with these changes needed for the local tests?

**Comment by howardjohn** [comment]
/retest

prow is stuck
